opendevin/runtime/docker/image_agnostic_util.py

```diff
@@ -0,0 +1,94 @@
+import tempfile
+
+import docker
+
+from opendevin.core.logger import opendevin_logger as logger
+
+
+def generate_dockerfile_content(base_image: str) -> str:
+    """
+    Generate the Dockerfile content for the agnostic sandbox image based on user-provided base image.
+
+    NOTE: This is only tested on debian yet.
+    """
+    # FIXME: Remove the requirement of ssh in future version
+    dockerfile_content = (
+        f'FROM {base_image}\n'
+        'RUN apt update && apt install -y openssh-server wget sudo\n'
+        'RUN mkdir -p -m0755 /var/run/sshd\n'
+        'RUN mkdir -p /opendevin && mkdir -p /opendevin/logs && chmod 777 /opendevin/logs\n'
+        'RUN wget "https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-$(uname)-$(uname -m).sh"\n'
+        'RUN bash Miniforge3-$(uname)-$(uname -m).sh -b -p /opendevin/miniforge3\n'
+        'RUN bash -c ". /opendevin/miniforge3/etc/profile.d/conda.sh && conda config --set changeps1 False && conda config --append channels conda-forge"\n'
+        'RUN echo "export PATH=/opendevin/miniforge3/bin:$PATH" >> ~/.bashrc\n'
+    ).strip()
+    return dockerfile_content
+
+
+def _build_sandbox_image(
+        base_image: str, target_image_name: str, docker_client: docker.DockerClient
+):
+    try:
+        with tempfile.TemporaryDirectory() as temp_dir:
+            dockerfile_content = generate_dockerfile_content(base_image)
+            logger.info(f'Building agnostic sandbox image: {target_image_name}')
+            logger.info(
+                (
+                    f'===== Dockerfile content =====\n'
+                    f'{dockerfile_content}\n'
+                    f'==============================='
+                )
+            )
+            with open(f'{temp_dir}/Dockerfile', 'w') as file:
+                file.write(dockerfile_content)
+
+            image, logs = docker_client.images.build(
+                path=temp_dir, tag=target_image_name
+            )
+
+        for log in logs:
+            if 'stream' in log:
+                print(log['stream'].strip())
+
+        logger.info(f'Image {image} built successfully')
+    except docker.errors.BuildError as e:
+        logger.error(f'Sandbox image build failed: {e}')
+        raise e
+    except Exception as e:
+        logger.error(f'An error occurred during sandbox image build: {e}')
+        raise e
+
+
+def _get_new_image_name(base_image: str) -> str:
+    if ":" not in base_image:
+        base_image = base_image + ":latest"
+
+    [repo, tag] = base_image.split(':')
+    return f'od_sandbox:{repo}_{tag}'
```

Comment:
Can we use a more robust deliminater between repo and tag? Say if the `repo` is `a_b_c` and tag is `d_e`, the whole string would be `a_b_c_d_e` which will be a bit hard to parse the `repo` and `tag`. How about we do something like: `{repo}__{tag}`?

Reply:
Got it. Although I don't recall any repo using underscores, using double underscores can indeed avoid potential issues. I will make the modification.

Reply:
Fixed it, and I also fixed the issue of mamba not working under the opendevin user. Please rerun the test.

Reply:
I rerun the test